i3pystatus/network.py

```diff
@@ -369,15 +377,38 @@ def run(self):
             network_usage = self.network_traffic.get_usage(self.interface)
             format_values.update(network_usage)
             if self.graph_type == 'input':
-                kbs = network_usage['bytes_recv']
+                limit = self.recv_limit
+                kbs = network_usage['bytes_recv'] * self.divisor / 1024
             elif self.graph_type == 'output':
-                kbs = network_usage['bytes_sent']
+                limit = self.sent_limit
+                kbs = network_usage['bytes_sent'] * self.divisor / 1024
             else:
                 raise Exception("graph_type must be either 'input' or 'output'!")
 
-            format_values['network_graph'] = self.get_network_graph(kbs)
-            format_values['kbs'] = "{0:.1f}".format(round(kbs, 2)).rjust(6)
-            color = self.get_gradient(kbs, self.colors, self.upper_limit)
+            format_values['network_graph'] = self.get_network_graph(kbs, limit)
+            format_values['kbs'] = "{0:.1f}".format(round(kbs, 2))
+
+            if self.separate_color and self.pango_enabled:
+                color = self.color_up
+                per_recv = network_usage["bytes_recv"] * self.divisor / (self.recv_limit * 1024)
+                per_sent = network_usage["bytes_sent"] * self.divisor / (self.sent_limit * 1024)
+                c_recv = self.get_gradient(int(per_recv * 100), self.colors, 100)
+                c_sent = self.get_gradient(int(per_sent * 100), self.colors, 100)
+                format_values["bytes_recv"] = "<span color=\"" + c_recv + "\">" + \
```

Comment:
This could be tidied up, how about something like:

                color_template = "<span color=\"{}\">{}</span>"
                format_values["bytes_recv"] = color_template.format(c_recv, network_usage["bytes_recv"])
                format_values["bytes_recv"] = color_template.format(c_sent, network_usage["bytes_sent"])